### ISSUE TITLE
type-annotation-spacing fails for "as T" syntax

### ISSUE THREAD

**Issue by nevir** [nzakas/eslint-plugin-typescript#42]
`type-annotation-spacing` doesn't play nice with `as T` type assertions:

```ts
const foo = {} as Foo;
//            ^ Unexpected space before the colon.
```

**Comment by weirdpattern** [comment]
This seems like another of those edge cases for the TypeAnnotation handler.
I will review this tomorrow and include it in the overrides that I just added in PR #43

**Comment by nevir** [comment]
Thanks!

**Comment by weirdpattern** [comment]
hey @JamesHenry you can close this bug too, it was merged in #43
